src/library/basecoverartdelegate.cpp

```diff
@@ -131,7 +131,7 @@ void BaseCoverArtDelegate::paintItem(
             } else {
                 // If we asked for a non-cache image and got a null pixmap,
                 // then our request was queued.
-                m_pendingCacheRows.insert(coverInfo.hash, index.row());
+                m_pendingCacheRows.insert(coverInfo.cacheKey(), index.row());
```

Comment:
Is this done in another thread that is already running? If so, could there be a race condition on very fast drives where the color is painted *after* the image? Would it make sense to move the `if (coverInfo.color)` part above line 113?

Reply:
No need to, perfectly safe. This code runs in the UI thread and values are not shared with any other thread.

A repaint is triggered by invalidating cells when the cover image becomes available.